src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1525,6 +1527,24 @@ private function dumpValue($value, $interpolate = true)
             }
 
             return sprintf('array(%s)', implode(', ', $code));
+        } elseif ($value instanceof LocatorArgument) {
+            $services = $value->getValues();
+            $servicesMap = array();
+            $getterCode = 'function ($id) {'."\n";
+            foreach ($services as $k => $ref) {
+                $servicesMap[] = sprintf("'%s' => '%s'", is_string($k) ? $k : $ref, $ref);
+                $singleGetter = sprintf('            if (\'%s\' === $id) {', is_string($k) ? $k : $ref);
+                $singleGetter .= sprintf("\n               return %s;", $this->getServiceCall((string) $ref, $ref));
+                $singleGetter .= "\n            }\n";
+                $getterCode .= $singleGetter;
+            }
+            $getterCode .= '        }';
+            $hasserCode = 'function ($id) {'."\n";
```

Comment:
given that keys should be the only way to reference values, instead of hasser+getter, I'd suggest generating a single `Closure[]`, keyed by "public" service name